### PR TITLE
Fix the deferred logic to not override the failed statements caught apart from deferred statements

### PR DESCRIPTION
**Pull request by priyanshi-yb**
**Issue:**
In case a few statements can't be deferred have issues, there is a bug in the current deferred logic which is overriding the non-deferred statements with just the deferred ones before putting them in the `failed.sql` while using `--continue-on-error true`.
https://yugabyte.atlassian.net/browse/DB-11653

**Fix:** 
Fixed the logic not to override `failedSqlStmts` where we put the failed statements non-deferred ones and only appending the final deferred statements that can't be executed in the `failedSqlStmts`.

**Tests:**
Manual testing is done, can't add automation for `continue-on-error` right now. In the future, we can have a test for `continue-on-error` and validate the `failed.sql` for expected failures. cc: @shubham-yb

### CODE REVIEW

File: yb-voyager/cmd/importSchemaYugabyteDB.go [hunk 1]
```diff
@@ -89,10 +91,11 @@ func importDeferredStatements() {
 		} else if beforeDeferredSqlCount == afterDeferredSqlCount {
 			// no need for further iterations since the deferred list will remain same
 			log.Infof("none of the deferred statements executed successfully in the %d iteration", i)
-			failedSqlStmts = failedSqlStmtInIthIteration
+			finalFailedDeferredStmts = failedSqlStmtInIthIteration
 			break
 		}
 	}
+	failedSqlStmts = append(failedSqlStmts, finalFailedDeferredStmts...)
```
Comment (sanyamsinghal) [inline yb-voyager/cmd/importSchemaYugabyteDB.go:98]: how about renaming global variable: `failedSqlStmts` -> `finalFailedSqlStmts`

File: yb-voyager/cmd/importSchemaYugabyteDB.go [hunk 2]
```diff
@@ -22,8 +22,9 @@ import (
 	"strings"
 
 	log "github.com/sirupsen/logrus"
-	"github.com/yugabyte/yb-voyager/yb-voyager/src/utils"
 	"golang.org/x/exp/slices"
+
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/utils"
```
Comment (sanyamsinghal) [inline yb-voyager/cmd/importSchemaYugabyteDB.go:27]: remove extra newline.

**Review by sanyamsinghal — APPROVED** [review]
LGTM, good catch and thanks for fixing.

File: yb-voyager/cmd/importSchemaYugabyteDB.go (hunk 2 above)
Comment (priyanshi-yb) [reply yb-voyager/cmd/importSchemaYugabyteDB.go:30]: its automatically added by my editor's go formatter

**Pull request merged**